### PR TITLE
Issue #196 - Remove old FF workaround setTimeout

### LINKED ISSUE #196: Test whether auto-select retry is still needed with Firefox >= 60
I've found these comments:

https://github.com/rugk/offline-qr-code/blob/e900c8878f86f2b3bea5558b686097b766fa9cba/src/popup/modules/UserInterface.js#L191

https://github.com/rugk/offline-qr-code/blob/e900c8878f86f2b3bea5558b686097b766fa9cba/src/popup/modules/UserInterface.js#L162

The code is responsible for selecting all the text and scrolling to the top when you open the QR code popup.

The comments indicate this is a workaround only for Firefox <= 60, but unfortunately no bug is linked or so.

However, nowadays we require Firefox 60 as a minimum version, anyway.

## TODO

* So we need to test – if we remove these workarounds –, does it still work in Firefox >= 60?
* If so, remove it from the code. (and clean up related code, if needed)

### PR DESCRIPTION
**Pull request by cornu-ammonis**
**Fixes #196** 

Confirmed via testing that this does not change add-on behavior. Also ran all tests. Bugzilla seems to [confirm](https://bugzilla.mozilla.org/show_bug.cgi?id=1324255) that the target of this workaround was resolved with FF 60.

### CODE REVIEW

File: src/popup/modules/UserInterface.js [hunk 1]
```diff
@@ -184,13 +174,7 @@ function scrollToTop(event) {
         return;
     }
 
-    // Attention: make sure this does not collide with the retry-property set
-    // in selectAllText()!
     event.setScrolled = true;
```
Comment (rugk) [inline src/popup/modules/UserInterface.js]: If we don't even re-try scrolling to the top, I guess we can remove all that `setScrolled` logic too, can't we?

**Comment by rugk** [comment]
BTW you can (automatically) [let issues close](https://help.github.com/articles/closing-issues-via-commit-messages/) when a PR is merged with the right keyword (i.e. remove "issue" in your PR description).

File: src/popup/modules/UserInterface.js (hunk 1 above)
Comment (cornu-ammonis) [reply src/popup/modules/UserInterface.js]: @rugk I agree! But I think there is still a setTimeout for that one [here](https://github.com/cornu-ammonis/offline-qr-code/blob/da9bdeb100dbb3cf028c19ba71dbb3b862a1c2e8/src/popup/modules/UserInterface.js#L153), should we take that out too?

**Comment by cornu-ammonis** [comment]
> Also `SELECT_TEXT_TIMEOUT` is set, but unused now. As per our contributing guide, try to use ESLint.

Ah my bad, will do! Excited to contribute!

File: src/popup/modules/UserInterface.js (hunk 1 above)
Comment (rugk) [reply src/popup/modules/UserInterface.js]: That's a good question and I am not sure. Is this still needed? Possibly not…?

Or possibly it is, considering "scroll to top" does not work anyway always in the current situation in my testing at least.

**Comment by cornu-ammonis** [comment]
> So I've also changed some things to fix that the selection was not properly triggered when you use the context menu in the add-on.
> 
> Also another issue:
> When you click inside of the input field, it does not always select the whole text automatically:
> gifSelectionIssue
> 
> There is an event listener that should force it to select all the text then: https://github.com/rugk/offline-qr-code/pull/199/files#diff-817dd83d26bd35283a14722780bb8481R516
> 
> Again, it seems this issue is present in the current master, anyway, so it may not be solvable in this PR.

Hmm yeah, does seem like some race stuff. I agree probably out of scope for this PR, but happy to file another issue and see if I can figure it out! Thanks for the comments / additions here!

**Comment by cornu-ammonis** [comment]
> Also `SELECT_TEXT_TIMEOUT` is set, but unused now. As per our contributing guide, try to use ESLint.

removed!

File: src/popup/modules/UserInterface.js (hunk 1 above)
Comment (cornu-ammonis) [reply src/popup/modules/UserInterface.js]: @rugk yeah, seems like that may call for a larger refactor

**Comment by rugk** [comment]
> happy to file another issue and see if I can figure it out! 

Done: https://github.com/rugk/offline-qr-code/issues/202

File: src/popup/modules/UserInterface.js (hunk 1 above)
Comment (rugk) [reply src/popup/modules/UserInterface.js]: Okay, fine, accepted. Also opened a different issue for that now: https://github.com/rugk/offline-qr-code/issues/203

**Comment by rugk** [comment]
Ah, BTW, next time, try to avoid doing a pull request from the master branch, because you can run into problems when you have a "non-clean" master that does not follow this repo here (i.e. "upstream"). [See this article for details](https://help.github.com/en/articles/proposing-changes-to-your-work-with-pull-requests).

**Comment by rugk** [comment]
BTW, currently there is the new [Hacktoberfest 2019](https://hacktoberfest.digitalocean.com/) running. Any PR there counts (if it is made in October). :smiley: 
So if you still want to figure out the reason for #202, maybe now is the time, @cornu-ammonis… :wink: 

You may also have a look at [all issues tagged with "hacktoberfest" in this repo](../issues/labels/hacktoberfest) or [somewhere else](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+archived%3Afalse+user%3Arugk+user%3ATinyWebEx+label%3A%22hacktoberfest%22+).

**Pull request merged**